Makefile

```diff
@@ -1,185 +1,84 @@
-endif
-
-# test if compiler is installed
-ifeq ($(shell which $(CC)), )
-    $(error compiler $(CC) is not installed)
-endif
-
-LL:=$(CC)
-
-# auto-discover all sources
-SRCS := $(shell find $(SRCS_DIR) -name '*.cpp')
-OBJS := $(patsubst $(SRCS_DIR)%.cpp,$(OBJS_RELEASE_DIR)%.o,$(SRCS))
-TEST_OBJS :=  $(patsubst $(SRCS_DIR)%.cpp,$(OBJS_DEBUG_DIR)%.o,$(SRCS))
-MAIN_OBJ := $(patsubst run/%.cpp, $(OBJS_RELEASE_DIR)/%.o, $(MAIN))
-
-DEP_DIR_RELEASE := $(OBJS_RELEASE_DIR)
-DEPFLAGS_RELEASE = -MT $@ -MMD -MP -MF $(DEP_DIR_RELEASE)/$*.d
-
-DEP_DIR_DEBUG := $(OBJS_DEBUG_DIR)
-DEPFLAGS_DEBUG = -MT $@ -MMD -MP -MF $(DEP_DIR_DEBUG)/$*.d
-
-INC_DIR := -I$(SRCS_DIR) -I$(LIBS_DIR)
-
-# run "make VERBOSE=1" to see all commands
-ifndef VERBOSE
-	Verb := @
-endif
-Echo := $(Verb)echo
-
-default: lint $(TARGET)
-
-$(TARGET): $(OBJS) $(MAIN_OBJ)
-	$(Echo) "Linking executable $(MAIN) into $@"
-	$(Verb) $(LL)  -o $@ $(OBJS) $(MAIN_OBJ) $(LFLAGS)
-
-$(MAIN_OBJ): $(OBJS_RELEASE_DIR)/%.o: $(MAIN)
-ifeq (,$(wildcard $(MAIN)))
-$(error file $(MAIN) does not exist)
-else
-	$(Echo) "Compiling $<"
-	$(Verb) mkdir -p $(dir $@)
-	$(Verb) $(CC) $(DEPFLAGS_RELEASE) $(CFLAGS) -o $@ -c $(INC_DIR) $<
-endif
-
-$(OBJS): $(OBJS_RELEASE_DIR)/%.o: $(SRCS_DIR)/%.cpp $(DEPENDENCIES)
-	$(Echo) "Compiling $<"
-	$(Verb) mkdir -p $(dir $@)
-	$(Verb) $(CC) $(DEPFLAGS_RELEASE) $(CFLAGS) -o $@ -c $(INC_DIR) $<
-
-
-$(TEST_OBJS): $(OBJS_DEBUG_DIR)/%.o: $(SRCS_DIR)/%.cpp $(DEPENDENCIES)
-		$(Echo) "Compiling $<"
-		$(Verb) mkdir -p $(dir $@)
-		$(Verb) $(CC) $(DEPFLAGS_DEBUG) $(CFLAGS) -o $@ -c $(INC_DIR) $< ${COVERAGE_FLAGS}
 lint:
-ifeq ($(NOLINT), 0)
+ifeq ($(RUNLINTER), 1)
 	$(Verb) python2.7 utils/Lint/presubmit.py --workspace=src
+else ifeq ($(PYTHONCHECK), 0)
+	$(error Cannot find Python 2.7, please check Python Version 2.7.X  is installed)
 endif
 
 lintall:
-	$(Echo) "\nLinting src/"
+ifeq ($(RUNLINTER), 1)
+	$(Echo) -e "\nLinting src/"
 	$(Verb) -python2.7 utils/Lint/presubmit.py --workspace=src
-	$(Echo) "\nLinting run/"
+	$(Echo) -e "\nLinting run/"
 	$(Verb) -python2.7 utils/Lint/presubmit.py --workspace=run
-	$(Echo) "\nLinting test/"
-	$(Verb) $(MAKE) -C test lint --no-print-directory
-
-static:
-	$(Verb) $(MAKE) -C test staticcheck CPPCHECK_ENABLE_OPS=$(STATIC_ENABLE)
-
-testrunner: test/lib/libtest.a
-	$(VERB) $(MAKE) -C test runner
-
-testrunner-all: test/lib/libtest.a
-	$(VERB) $(MAKE) -C test runnerall 
-
-testrunner-filter: test/lib/libtest.a
-	$(VERB) $(MAKE) -C test runnerfilter GOOGLE_TEST_FILTERS=$(GTEST_FILTERS)
-
-coverage: testrunner
-	$(Verb) ./test/utils/get_code_cov.sh
-
-test/lib/libtest.a:  $(EIGEN) $(RAPIDJSON) $(TEST_OBJS)
-	$(Echo) "Making library"
-	$(Verb) ar -cvr $@ $(TEST_OBJS) > /dev/null
+	$(Echo) -e "\nLinting test/"
+	$(Verb) -python2.7 utils/Lint/presubmit.py --workspace=test
+else ifeq ($(PYTHONCHECK), 0)
+	$(error Cannot find Python 2.7, please check Python Version 2.7.X  is installed )
+endif
 
-clean-all: cleanlint cleantest clean
+clean-all: clean
```

Comment:
What is the point of this target, and what was the reason for the removal of `cleanlint` and `cleantest`?